docarray/array/doc_dict/doc_dict.py

```diff
@@ -0,0 +1,131 @@
+from typing import TYPE_CHECKING, Any, Dict, Iterable, Mapping, Type, TypeVar, Union
+
+import orjson
+from typing_inspect import is_union_type
+
+from docarray.array.any_collections import AnyCollection
+from docarray.array.doc_list.doc_list import DocList
+from docarray.base_doc import AnyDoc
+from docarray.base_doc.doc import BaseDoc
+from docarray.base_doc.io.json import orjson_dumps
+
+if TYPE_CHECKING:
+    from docarray.proto import DocDictProto
+    from docarray.typing import NdArray, TorchTensor
+
+T_doc = TypeVar('T_doc', bound=BaseDoc)
+T = TypeVar('T', bound='DocDict')
+
+
+class DocDict(AnyCollection[T_doc], Dict[str, T_doc]):
+    doc_type: Type[BaseDoc] = AnyDoc
+
+    def __init__(self, **mapping):
+        super().__init__(
+            {key: self._validate_one_doc(val) for key, val in mapping.items()}
+        )
+
+    def _validate_one_doc(self, doc: T_doc) -> T_doc:
+        """Validate if a Document is compatible with this `DocList`"""
+        if not issubclass(self.doc_type, AnyDoc) and not isinstance(doc, self.doc_type):
+            raise ValueError(f'{doc} is not a {self.doc_type}')
+        return doc
+
+    @classmethod
+    def from_doc_list(cls: Type[T], docs: Union[DocList, Iterable[BaseDoc]]) -> T:
```

Comment:
Could this just happen in the constructor? Also, if an iter of docs is accepted, then the name is a bit misleading

Reply:
> Also, if an iter of docs is accepted, then the name is a bit misleading

yep I had the same feeling.

How should I named it them ?

1- `from_docs`
2- `from_seq_docs`
3- ???

Reply:
`from_docs` is good I think